### PR TITLE
rio: Restrict test execution inside nixosTests for Linux devices

### PR DESCRIPTION
**Pull request by otavio**
-------

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by TornaxO7 — APPROVED** [review]

File: pkgs/applications/terminal-emulators/rio/default.nix [hunk 1]
```diff
@@ -111,8 +111,9 @@ rustPlatform.buildRustPackage rec {
       extraArgs = [ "--version-regex" "v([0-9.]+)" ];
     };
 
-    tests = {
-      test = nixosTests.terminal-emulators.rio;
+    tests = lib.optionalAttrs stdenv.isLinux {
```
Comment (mattpolzin) [inline pkgs/applications/terminal-emulators/rio/default.nix:114]:
```suggestion
    tests = lib.optionalAttrs stdenv.buildPlatform.isLinux {
```
There was a (very recent) push to be more specific about what environment's platform is being checked with `isX` calls so cross compilation is at less of a deficit.

**Comment by NickCao** [comment]
This affects practically every package with a linked nixos test, can we find a solution for them all?

**Comment by otavio** [comment]
This would be for sure the best solution, however I don't have access for the Darwin. So, I believe that someone using this operating system will need to work on that.

Meanwhile, I think that this should not forbid the value of this pull request, as this solves a real problem and other derivations do the same already.

**Comment by NickCao** [comment]
> Meanwhile, I think that this should not forbid the value of this pull request, as this solves a real problem and other derivations do the same already.

Could you open a tracking issue (if there isn't one already) and link to it in a comment. So we can just grep and replace all the occurrences of this hack when the underlying issue is fixed.

**Comment by NickCao** [comment]
The PR that introduced this problem: https://github.com/NixOS/nixpkgs/pull/303597

**Comment by otavio** [comment]
@NickCao Honestly, I don't have too much detail about the Darwin issue, so it would be better if someone with more involvement in Darwin can report that. I'm totally fine to update the code and the commit log to mention this issue so it can be searched in the future.

Don't get me wrong, I'm fine in reporting issues, just I don't have the details to be able to do a proper issue with helpful information.

@mattpolzin Could you help on that?

**Comment by NickCao** [comment]
Also cc @wegank

**Comment by mattpolzin** [comment]
Sure, I can write up a ticket. I actually didn't realize this was related to only recently starting to run the nixos tests on darwin at all; thanks for the link Nick.

**Review by mattpolzin — APPROVED** [review]
Thanks. We may as well let the aarch64-darwin test play out (though clearly things are being skipped as desired for the already-finished x86_64-darwin passthru tests) before merging.

**Pull request merged**